core/src/avm2/activation.rs

```diff
@@ -1484,7 +1496,7 @@ impl<'a, 'gc> Activation<'a, 'gc> {
                         name_value.as_object().unwrap(),
                         self.context.gc_context,
                     );
-
+                    object.remove_shape_id(self.context.gc_context);
```

Comment:
Please double check that does slot id change or not upon delete property bytecode.